File: benchmark/src/bench.py
def fib(n):
    return n if n <= 1 else fib(n-1)+fib(n-2)

def sieve(n):
    p = [True]*(n+1); p[0]=p[1]=False
    for i in range(2, int(n**0.5)+1):
        if p[i]: p[i*i:n+1:i] = [False]*((n - i*i)//i + 1)
    return sum(p)

print(f"fib(35) = {fib(35)}")
print(f"sieve(500000) = {sieve(500000)} primes")

File: benchmark/src/bench.go
package main
import "fmt"
func fib(n int64) int64 { if n <= 1 { return n }; return fib(n-1) + fib(n-2) }
func sieve(n int) int {
    p := make([]bool, n+1); for i := 0; i <= n; i++ { p[i] = true }; p[0], p[1] = false, false
    for i := 2; i*i <= n; i++ { if p[i] { for j := i * i; j <= n; j += i { p[j] = false } } }
    c := 0; for _, v := range p { if v { c++ } }; return c
}
func main() {
    fmt.Printf("fib(35) = %d\n", fib(35))
    fmt.Printf("sieve(500000) = %d primes\n", sieve(500000))
}
